src/josepy/json_util.py

```diff
@@ -409,17 +412,17 @@ def decode_csr(b64der):
 class TypedJSONObjectWithFields(JSONObjectWithFields):
     """JSON object with type."""
 
-    typ = NotImplemented
+    typ = NotImplemented  # type: Any
```

Comment:
Without this, there are quite a lot of warnings about `_NotImplementedType` in the subclasses.